rotkehlchen/db/utils.py

```diff
@@ -309,11 +319,10 @@ def replace_tag_mappings(
     insert_tuples = []
     delete_tuples = []
     for entry in data:
+        # Always delete current tags even if there are no new tags.
+        delete_tuples.append((_get_tag_reference(entry, object_reference_keys),))
```

Comment:
Could you please add a test to show what added benefit/improvement it brings?
Really difficult to know what side effects it introduces

Reply:
Yes, makes sense, for some reason thought there already is one. Will add.

Reply:
I don't think there is a need in a test since it is just a refactor. It does not fix anything, but prevents us from hitting errors like the one fixed in #5538.

Reply:
oh okay then.